### DISCUSSION TITLE
Moisture capacitive sensor

### DISCUSSION THREAD

**Discussion by paulzhdanov** [RavenSystem/esp-homekit-devices#1537] [General]
Hi, could you please help with the scheme and json of how to use properly moisture sensor with analogue output? I mean non binary reaction, it works fine with comparator. I'd like to measure soil moisture in %% 'cause binary reaction set by potentiometer is too rough.

What I already made:
1 simple scheme with comparator and A0 output connected to A0 at NodeMCU. t:23, n:9, doesn't work
2 scheme without comparator but with capacitive soil moisture sensor v2.0, doesn't work (screenshot)
3 scheme with ADS1115 connected to SCL/SDA pins from one side and capacitive soil moisture sensor v2.0, JSON: n:4 (i2c), doesn't work...

Please advice how to do it properly?

![Capacitive-Soil-Moisture-Sensor-ESP8266 diagram](https://user-images.githubusercontent.com/13831213/131178280-75ea9034-f223-4b53-aa3b-457588c28e03.JPG)

Version 6.10.0
JSON:
{
  "c":{"n":"haahum"},
  "a": [{
    "t": 23,
    "n": 9,
    "j": 1
  }]
}

**Comment by RandomPotato893** [comment]
I have this same question, did you figure it out

**Reply by paulzhdanov** [reply]
No. I made several configs and hardware versions including i2c (through mcp23017) but it doesn't work. No idea how to make it. The only option is leak sensor, useful in the house, but useless for irrigation, I need %% of moisture...

**Comment by mario-website** [comment]
I would like to know too :) @RavenSystem maybe you know something about how to fixed/set correctly? I saw few post in past related to soil moisture.

**Comment by paulzhdanov** [comment]
@RavenSystem Probably free monitor may help with this? Unfortunately I didn't find how, please advice.
As a sample I use that script just to try:

{
  "a":[{
      "t":23,
      "es":[{
        "t":80,
        "n":10,
        "h":0,
        "y0": [{
            "v":300,
            "0":{ "m":[[-1,10]]}
          },{
            "v":500,
            "0":{ "m":[[-1,20]]}
          },{
            "v":600,
            "0":{ "m":[[-1,30]]}
          },{
            "v":700,
            "0":{ "m":[[-1,40]]}
          },{
            "v":800,
            "0":{ "m":[[-1,50]]}
        }]
      }]
  }]
}

**Comment by niquini** [comment]
Has anyone managed to get this to work?

**Reply by paulzhdanov** [reply]
Unfortunately no, the only thing I managed to do is to create simple humidity sensor (2 wires stick to the ground) connected to ADC. It's not very accurate (max value I got was ~75%), but better, than nothing. But you may look at tasmota firmware, it has chirp driver (not included in default config) and matter option to connect.
